src/trajectory/Interpolated.cpp

```diff
@@ -210,5 +217,13 @@ bool Interpolated::Waypoint::operator<(double rhs) const
   return t < rhs;
 }
 
+//==============================================================================
+void Interpolated::Waypoint::free(
+    const statespace::ConstStateSpacePtr& stateSpace)
+{
+  if (state)
+    stateSpace->freeState(state);
```

Comment:
I just realized that this could double-free when this function is called multiple times. Let's prevent it by setting `state` to `nullptr` once it's freed as:
```cpp
if (state)
{
  stateSpace->freeState(state);
  state = nullptr;
}
```

Reply:
Oh, good catch! :)